Jenkinsfile

```diff
@@ -1,21 +1,61 @@
 elifeLibrary {
+    def isNew
+    def candidateVersion
     def commit
+
     stage 'Checkout', {
         checkout scm
         commit = elifeGitRevision()
     }
 
-    stage 'Build image', {
-        sh 'docker build -t elife/sciencebeam-alignment .'
-    }
+    node('containers-jenkins-plugin') {
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+            try {
+                candidateVersion = sh(
+                    script: (
+                        "IMAGE_TAG=${commit} " +
+                        "docker-compose -f docker-compose.yml -f docker-compose.ci.yml run " +
+                        "sciencebeam-alignment ./print_version.sh"
+                    ),
+                    returnStdout: true
+                ).trim()
+                echo "Candidate version: v${candidateVersion}"
+            } finally {
+                sh 'docker-compose down -v'
+            }
+        }
+
+        stage 'Project tests', {
+            try {
+                sh "IMAGE_TAG=${commit} " +
+                    "docker-compose -f docker-compose.yml -f docker-compose.ci.yml " +
+                    "run sciencebeam-alignment ./project_tests.sh"
+            } finally {
+                sh 'docker-compose down -v'
```

Comment:
the `docker-compose down` command should still use the same `-f` configuration as the other commands, but we'll solve this with the command extraction